File: src/lib/tauri-api/agent-runs.ts
import type { StartWorkspaceRunInput, WorkspaceRun, WorkspaceRunLog } from '../../types/agent-run';
import { invokeCommand } from './client';

export function startWorkspaceRun(input: StartWorkspaceRunInput): Promise<WorkspaceRun> {
  return invokeCommand<WorkspaceRun>('start_workspace_run', { input });
}

export function stopWorkspaceRun(runId: string): Promise<WorkspaceRun> {
  return invokeCommand<WorkspaceRun>('stop_workspace_run', { runId });
}

export function getWorkspaceRuns(workspaceId: string): Promise<WorkspaceRun[]> {
  return invokeCommand<WorkspaceRun[]>('get_workspace_runs', { workspaceId });
}

export function getWorkspaceRunLogs(runId: string): Promise<WorkspaceRunLog[]> {
  return invokeCommand<WorkspaceRunLog[]>('get_workspace_run_logs', { runId });
}

File: frontend/src-tauri/src/commands/agent_runs.rs
use tauri::State;

use crate::models::{StartWorkspaceRunInput, WorkspaceRun, WorkspaceRunLog};
use crate::services::agent_process_service;
use crate::state::AppState;

#[tauri::command]
pub fn start_workspace_run(
    state: State<'_, AppState>,
    input: StartWorkspaceRunInput,
) -> Result<WorkspaceRun, String> {
    agent_process_service::start_workspace_run(&state, input)
}

#[tauri::command]
pub fn stop_workspace_run(
    state: State<'_, AppState>,
    run_id: String,
) -> Result<WorkspaceRun, String> {
    agent_process_service::stop_workspace_run(&state, &run_id)
}

#[tauri::command]
pub fn get_workspace_runs(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<Vec<WorkspaceRun>, String> {
    agent_process_service::get_workspace_runs(&state, &workspace_id)
}

#[tauri::command]
pub fn get_workspace_run_logs(
    state: State<'_, AppState>,
    run_id: String,
) -> Result<Vec<WorkspaceRunLog>, String> {
    agent_process_service::get_workspace_run_logs(&state, &run_id)
}
